src/Test/L0/Worker/TestResults/Mocks/MockTestManagementHttpClient.cs

```diff
@@ -0,0 +1,353 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.TeamFoundation.TestManagement.WebApi;
+using Microsoft.VisualStudio.Services.Agent.Worker.TestResults;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.VisualStudio.Services.Agent.Tests.Worker.TestResults
+{
+    public class MockTestManagementHttpClient : ITestManagementHttpClient
```

Comment:
I guess this is some testable class, this has lots of notimplementedexception. If you have a ServiceLocator, this class can go away and you can setup which ever method is required in your corresponding tests